app/src/main/java/com/example/coctailapp/ui/screens/main/content/cocktails/CocktailsContentViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.coctailapp.ui.screens.main.content.cocktails
+
+import android.content.Context
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import coil3.network.HttpException
+import com.example.coctailapp.R
+import com.example.coctailapp.model.CocktailsPreview
+import com.example.coctailapp.network.CocktailsApi
+import dagger.hilt.android.lifecycle.HiltViewModel
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import okio.IOException
+import javax.inject.Inject
+
+
+@HiltViewModel
+class CocktailsContentViewModel @Inject constructor(private val cocktailsApi: CocktailsApi,
+    @ApplicationContext val context : Context): ViewModel() {
+
+    private var _cocktailsList = MutableStateFlow(emptyList<CocktailsPreview>())
```

Comment:
Can we remove this stateFlow? We should try to move the data list inside the fetchStatus event